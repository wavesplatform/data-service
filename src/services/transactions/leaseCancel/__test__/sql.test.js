const sql = require('../sql');
const filterValues = {
  // default values
  sort: 'desc',
  limit: 100,
  // for test
  recipient: 'recipient',
};

describe('Sql search by type-specific filters', () => {
  it('supports recipient filter', () => {
    expect(sql.search(filterValues)).toMatchSnapshot();
  });
});
