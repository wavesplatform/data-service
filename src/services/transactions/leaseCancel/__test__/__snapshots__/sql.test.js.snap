// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql search by type-specific filters supports recipient filter 1`] = `"with \\"t_cte\\" as (select \\"t\\".\\"tx_uid\\" as \\"tx_uid\\", \\"t\\".\\"height\\" as \\"height\\", \\"t\\".\\"sender_uid\\" as \\"sender_uid\\", \\"l\\".\\"id\\" as \\"lease_id\\" from \\"txs_9\\" as \\"t\\" left join \\"txs\\" as \\"l\\" on \\"l\\".\\"uid\\" = \\"t\\".\\"lease_tx_uid\\" where \\"lease_tx_uid\\" in (select \\"tx_uid\\" from \\"txs_8\\" where \\"recipient_address_uid\\" = (select \\"uid\\" from \\"addresses\\" where \\"address\\" = 'recipient' limit 1)) order by \\"t\\".\\"tx_uid\\" desc limit 100) select \\"t\\".\\"tx_uid\\" as \\"tx_uid\\", \\"t\\".\\"height\\" as \\"height\\", \\"txs\\".\\"tx_type\\" as \\"tx_type\\", \\"txs\\".\\"id\\" as \\"id\\", \\"txs\\".\\"time_stamp\\" as \\"time_stamp\\", \\"txs\\".\\"signature\\" as \\"signature\\", \\"txs\\".\\"proofs\\" as \\"proofs\\", \\"txs\\".\\"tx_version\\" as \\"tx_version\\", txs.fee * 10^(-8) as \\"fee\\", \\"addr\\".\\"address\\" as \\"sender\\", \\"addr\\".\\"public_key\\" as \\"sender_public_key\\", \\"t\\".\\"lease_id\\" as \\"lease_id\\" from \\"t_cte\\" as \\"t\\" left join \\"txs\\" on \\"txs\\".\\"uid\\" = \\"t\\".\\"tx_uid\\" left join \\"addresses\\" as \\"addr\\" on \\"addr\\".\\"uid\\" = \\"t\\".\\"sender_uid\\" order by \\"txs\\".\\"uid\\" desc limit 100"`;
